statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,276 @@
+"""
+Methods for creating models for survey data.
+
+The main classes are:
+
+  * SurveyModel : Creates and fits the specified model. Returns
+  estimates and stderr
+
+Note: This makes use of survey weights, which is supplied to
+the 'freq_weights' parameter in StatsModel's GLM. SurveyModel can
+only be utilized by models w/ a freq_weight or weight parameter
+"""
+
+from __future__ import division
+import numpy as np
+import statsmodels
+
+
+class SurveyModel(object):
+    """
+
+    Parameters
+    -------
+    design : Instance of class SurveyDesign
+    model_class : Instance of class GLM
+    init_args : Dictionary of arguments
+        when initializing the model
+    fit_args : Dictionary of arguments
+        when fitting the model
+
+    Attributes
+    ----------
+    design : Instance of class SurveyDesign
+    model : Instance of class GLM
```

Comment:
I think model is actually the model class, not an instance.  Docstring needs to be fixed, and perhaps also rename the argument to "cls"

Reply:
I like longer names like `model_class`. We just moved to `results_class` as attribute in discrete models (old name, chosen by me, was `result` which got me confused now, a few years after writing the code)